src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -244,9 +244,23 @@ def _get_string_datum(self, query: str) -> Optional[str]:
     @SnowflakePlan.Decorator.wrap_exception
     def get_result_attributes(self, query: str) -> List[Attribute]:
         return convert_result_meta_to_attribute(
-            run_new_describe(self._cursor, query), self.max_string_size
+            self._run_new_describe(self._cursor, query), self.max_string_size
         )
 
+    def _run_new_describe(
+        self, cursor: SnowflakeCursor, query: str
+    ) -> Union[List[ResultMetadata], List["ResultMetadataV2"]]:
+        result_metadata = run_new_describe(cursor, query)
+
+        for listener in filter(
+            lambda listener: hasattr(listener, "include_describe")
```

Comment:
I don't understand why we need a `hasattr` check here. isn't `listener.include_describe` sufficient?

Reply:
For example, in the case of SqlCounter we don't inherit from QueryHistory, we just implement _add_query directly on the class and add as a listener.  However, adding this property breaks the client unless the new property is added.  May be we could do this differently but this isn't an uncommon pattern so added the check here to avoid potentially breaking clients that do this kind of duck typing.